aim/sdk/adapters/mxnet.py

```diff
@@ -0,0 +1,156 @@
+        self.train_start = time.time()
+        trainer = estimator.trainer
+        optimizer = trainer.optimizer.__class__.__name__
+        lr = trainer.learning_rate
+
+        estimator.logger.info("Training begin: using optimizer %s "
+                              "with current learning rate %.4f ",
+                              optimizer, lr)
+        if estimator.max_epoch:
+            estimator.logger.info("Train for %d epochs.", estimator.max_epoch)
+        else:
+            estimator.logger.info("Train for %d batches.", estimator.max_batch)
+        # reset all counters
+        self.current_epoch = 0
+        self.batch_index = 0
+        self.processed_samples = 0
+        self.log_interval_time = 0
+
+        params = {
+            "arch": estimator.net.name,
+            "loss": estimator.loss.name,
+            "optimizer": optimizer,
+            "lr": lr,
+            "max_epoch": estimator.max_epoch,
+            "max_batch": estimator.max_batch
+        }
+
+        self.setup(estimator, params)
+
+    def train_end(self, estimator, *args, **kwargs):
+        train_time = time.time() - self.train_start
+        msg = 'Train finished using total %ds with %d epochs. ' % (train_time, self.current_epoch)
+        # log every result in train stats including train/validation loss & metrics
+        for metric in self.metrics:
+            name, value = metric.get()
+            msg += '%s: %.4f, ' % (name, value)
+        estimator.logger.info(msg.rstrip(', '))
+
+    def batch_begin(self, estimator, *args, **kwargs):
+        if isinstance(self.log_interval, int):
+            self.batch_start = time.time()
+
+    def batch_end(self, estimator, *args, **kwargs):
+        if isinstance(self.log_interval, int):
+            batch_time = time.time() - self.batch_start
+            msg = '[Epoch %d][Batch %d]' % (self.current_epoch, self.batch_index)
+            self.processed_samples += kwargs['batch'][0].shape[0]
+            msg += '[Samples %s] ' % (self.processed_samples)
+            self.log_interval_time += batch_time
+            if self.batch_index % self.log_interval == 0:
+                msg += 'time/interval: %.3fs ' % self.log_interval_time
+                self.log_interval_time = 0
+                for metric in self.metrics:
+                    # only log current training loss & metric after each interval
+                    name, value = metric.get()
+                    msg += '%s: %.4f, ' % (name, value)
+
+                    context_name, metric_name = name.split(" ")
+                    context = {'subset': context_name}
+                    self._run.track(value, metric_name, step=self.batch_index, context=context)
+                estimator.logger.info(msg.rstrip(', '))
+        self.batch_index += 1
+
+    @property
+    def experiment(self) -> Run:
+        if not self._run:
+            self.setup()
+        return self._run
+
+    def setup(self, estimator, args=None):
```

Comment:
`estimator` should be optional as well, as there's a call to `setup()` without any arguments

Reply:
Fixed the missing types and made `estimator` optional. I think having a `setup()` without arguments in the `experiment` property is normal, because the `estimator` is needed to extract the parameters.